### ISSUE TITLE
Support for Insta360 Link

### ISSUE THREAD

**Issue by pzich** [Insta360Develop/CameraSDK-Cpp#12]
The Insta360 Link is not listed as one of the supported cameras. Can it please be added?

**Comment by enfipy** [comment]
+1, also would really need this.

**Comment by rwebber** [comment]
I would also love this ASAP :)

**Comment by SimeonOA** [comment]
Has there been any development on this?

@pzich @enfipy @rwebber did any of you find a way to support the link?

Thank you.

**Comment by SNRSE** [comment]
+1 
this would really make the link more usefull

**Comment by ablindsay** [comment]
We very much need an SDK or we need to start considering a different camera solution (like OBSbot)

**Comment by jonfishr** [comment]
+1 for Insta360 Link camera

**Comment by marcelomoyano** [comment]
Agreed. This is a need.

**Comment by bowdeng** [comment]
+1 - would mean the Insta360 could be used in so many more pipelines

**Comment by HMarshallM** [comment]
As a new owner of the 360Link I would think it will benefit me if it was listed as one of the supported cameras.

**Comment by butram** [comment]
I would buy one tomorrow if the Insta360 Link had an SDK.

**Comment by statik** [comment]
I need this too please.

**Comment by jwnichols3** [comment]
This will make a big difference for my setup.

**Comment by peterrosado** [comment]
I would love to purchase more 360links however the lack of SDK to implement OSC controls limits me and my options. The osbot tiny2 has OSC controls but I want to stay inside the Insta family please; thank you!

**Comment by linggesar** [comment]
SDK for the Insta360 would make our buying decision much easier.

**Comment by eliaspuurunen** [comment]
+1 for our live event production needs.

**Comment by alangoldberg** [comment]
I agree with the comments above for the SDK

**Comment by paulbuchan** [comment]
Yes! Please add an SDK.

**Comment by joreste** [comment]
An SDK for the Insta360 Link would be greatly appreciated

**Comment by nubby0614** [comment]
Is like to see SDK support for the Insta360 Link as well.

**Comment by eyckp** [comment]
Please make an SDK for the Insta360 Link available rapidly. Otherwise I will go for some cameras of Obsbot.

**Comment by ElmsPark** [comment]
Add me to the list for an SDK for the Insta360 Link

**Comment by zerodistraction** [comment]
+1 for an SDK for the Insta360 Link so more controls can be added. This will allow many of us that send these out in remote kits to really up our production level.

**Comment by pbelbin** [comment]
In the case of the Link, SDK should support not only the exposure & configuration details, but also provide a way to allow remote control of PTZ functions.

**Comment by clafarge** [comment]
Must... have... sdk...

**Comment by 22architecture** [comment]
I’d like a way to reposition the AI tracking on my Insta360 link so the eyes are higher in the frame. Thanks

**Comment by craigmacdev** [comment]
yes, this is a big gap.  It isn't a big deal at launch time, but in today's world nothing is serious if it doesn't have an API and SDK

**Comment by njushchyshyn** [comment]
+1 Been desperately wanting this for a long time. Really love the image quality of this camera, but not being able to develop and integrate with other tools we’re using limits usefulness so we have been holding off on getting more cameras for our labs and studios.

**Comment by TroikaTronix** [comment]
+1 We would love to add PTZ control to an actor in our software Isadora which is used by thousands of artists worldwide. Please add the Insta360 to the supported cameras, and ensure that we can also control the movement of the camera, and we will quickly be making a module for our software to leverage these features.

**Comment by rmminnc** [comment]
+1 for an SDK for the Insta360 Link

**Comment by swy** [comment]
This would be great.

**Comment by WizyWizer** [comment]
I support SDK for insta360 Link...

**Comment by tgbauer** [comment]
+1. SDK would help here

**Comment by abruptriver** [comment]
+1 for an SDK. I could really use this.

**Comment by alexbuckland** [comment]
Please open this up for link. PTZ control via 3rd party is a must have for my workflow. Pweaaase.

**Comment by toddreyn** [comment]
OH yeah, this would be just awesome.  Would love an SDK!

**Comment by lightconversations** [comment]
+1. This would make this camera so much more usable!!

**Comment by TodR64** [comment]
+1, please add the SDK

**Comment by heII0w0rId** [comment]
An SDK for Insta360 devices would be appreciated. Insta360’s opportunity to capture new verticals for on-market products, alone is significantly degraded without an SDK. Cheers!

**Comment by RobinCutshaw** [comment]
So many cool things we could do if we had software control through an sdk.  I cringe to say it but, think what an AI interface could provide.

**Comment by DrMattyWood** [comment]
SDK really needed for insta360 - would solve a lot of issues

**Comment by mandy-van** [comment]
Please provide an SDK for the Insta360 Link.

**Comment by danielmcfarland** [comment]
Would be great to have an SDI for this.

**Comment by brettbelau** [comment]
How soon can Insta360 Link owners get SDK access??

**Comment by timdavidatmac** [comment]
I am interested in developers having access to an SDK for Link to better support people with disabilities, such as those that are blind/low vision, Deaf/HoH, and with mobility disabilities.

**Comment by mcardlek** [comment]
+1 for sdk

**Comment by s-vika** [comment]
+1 A sdk would make the Insta360 Link alot more compelling.

**Comment by TJP66** [comment]
We could use many of these cameras in remote kits if we had an SDK to build controls to access all of the camera's features.

**Comment by ronhof** [comment]
I need more control that what your existing software provide, compared to what the Insta360 Link could do with the power an sdk or api would provide. I have been contacting you before about API/SDK, i would love to extend the functionality of this amazing device.

**Comment by ASRW1** [comment]
Currently have these sitting on the shelf as I’m looking for the control I could build with an SDK like your other excellent cameras

**Comment by Woobrd** [comment]
+1 for an SDK

**Comment by leonardzh** [comment]
We need SDK for Link!

**Comment by pzich** [comment]
@zhangyabin @capjason @capjason-insta360 can we _please_ get support for this?

**Comment by FarCoded** [comment]
The Insta360 Link is not listed as one of the supported cameras. Can it please be added?

**Comment by Huseriato** [comment]
@capjason-insta360 Is there any deadline for this? I'm really waiting. The OBSBOT cameras do support OSC protocol (documentation here: https://www.obsbot.com/de/explore/obsbot-webcam/osc). I'm really missing support to enable and disable tracking via software and control the camera position. It's one of the needed main features.

**Comment by IainMNorman** [comment]
+1 from me. Being able to code up my own plugin for my streamdeck without resorting to global hotkeys would definitely push me to purchase.

**Comment by Sigtechnica** [comment]
+1 would be nice to have some cli control for the insta360 link!

**Comment by njushchyshyn** [comment]
If not a full API (which is VERY MUCH desired/preferred) even having some way to interact with the exisisting control software via a standard protocol like OSC would be very helpful in allowing tools like Bitfocus Companion and many other streaming tools to interact with this camera that is a really nice camera, but genuinely fails to reach its full potential without an API or external interaction abilities.

**Comment by mjlee111** [comment]
You can control insta 360 link with livuvc or v4l2 on linux(pan, tilt or etc).  https://github.com/mjlee111/v4l2_gui.git here is a simple cpp project .

**Comment by Huseriato** [comment]
> You can control insta 360 link with livuvc or v4l2 on linux(pan, tilt or etc). https://github.com/mjlee111/v4l2_gui.git here is a simple cpp project .

That is not an api. This just covers a small subset of potential commands. For example it does not allow to turn on or off follow mode or any other specific Insta360 function. The mentioned project is nothing special, OBS can do the same things with the move plugin.

Insta360 Link is still missing a real api. I think there will never be any api for this camera. I already switched to OBSBot and bought 3 cameras there.

**Comment by mjlee111** [comment]
> > You can control insta 360 link with livuvc or v4l2 on linux(pan, tilt or etc). https://github.com/mjlee111/v4l2_gui.git here is a simple cpp project .
> 
> That is not an api. This just covers a small subset of potential commands. For example it does not allow to turn on or off follow mode or any other specific Insta360 function. The mentioned project is nothing special, OBS can do the same things with the move plugin.
> 
> Insta360 Link is still missing a real api. I think there will never be any api for this camera. I already switched to OBSBot and bought 3 cameras there.

Thanks for the good advice. OBSBot's camera could be a good alternative. If I get a chance, I'll try developing the API of insta360 link as a subproject. Thank you.

**Comment by Sigtechnica** [comment]
@mjlee111 , I have tried it , it works quite nice. 
the moving is funny because the 'AI' also wants to move it , to track me :)

I prefer the insta360 because its 2-axis gimbal

**Comment by mjlee111** [comment]
@Sigtechnica Nice work! I really agree with preference of insta360. Can you shrare your work for others?

**Comment by Sigtechnica** [comment]
sorry for (very) late reply I did not do any changes, I just compiled it :)
